thrall/app/lib/elasticsearch/ElasticSearch.scala

```diff
@@ -27,6 +27,20 @@ class ElasticSearch(config: ElasticSearchConfig, metrics: Option[ThrallMetrics])
   lazy val shards = config.shards
   lazy val replicas = config.replicas
 
+  def bulkInsert(images: Seq[Image])(implicit ex: ExecutionContext): List[Future[ElasticSearchBulkUpdateResponse]] = {
+    val request = bulk {
+      images.map(img => {
+        indexInto(imagesAlias, Mappings.dummyType)
+          .id(img.id)
+          .source(Json.stringify(Json.toJson(img)))
+      })
+    }
+
```

Comment:
Suggestion to consider:

according to contract between lambda that `produces and notify`
the blob read from s3 will be json string with `\n` delimiter
so i think we do not need to serialize and deserialize that again

Unless you think that it will improve contract between services

Reply:
The intention here is for `bulkInsert` to only accept `Seq[Image]` as input for type safety.

The contract between the services will be upheld in the [message processor](https://github.com/guardian/grid/blob/master/thrall/app/lib/kinesis/MessageProcessor.scala#L22). On receipt of a message, `thrall` will get the JSON blob and attempt to parse it as `Seq[Image]` and call `bulkInsert` if parsing is successful, if unsuccessful, log and fail. This provides a nice separation of concern as the ES class doesn't care about validating the S3 JSON. Reading the JSON from S3 will be done in a different PR.

Reply:
yeah i think performance implication will be very small so i am ok with that